src/core/layertree/qgslayertreemodel.cpp

```diff
@@ -1527,7 +1527,6 @@ void QgsLayerTreeModel::invalidateLegendMapBasedData()
     Q_FOREACH ( QgsSymbolLegendNode *n, symbolNodes )
     {
       const QString parentKey( n->data( QgsLayerTreeModelLegendNode::ParentRuleKeyRole ).toString() );
-      Q_ASSERT( widthMax[parentKey] > 0 );
```

Comment:
Hi @vmora,

Do you have an important piece of information on the subject? Or can I remove the `Q_ASSERT` statement safely? Broadly speaking, I don't like to remove an assert (as Matthias has pointed out), but it's affecting our testing environment running in Docker containers (so within the virtual X server environment thanks to `xvfb-run`).

Let me know :).